### PR TITLE
Add px-app-route

### PR DESCRIPTION
**Pull request by davidrleonard**
The `px-app-route` component helps developers bind their app's URL with the app state (in components like `px-app-nav`, `px-context-browser`, etc.)

The documentation in `px-app-route/px-app-route.html` is very detailed and should explain exactly how developers can use this component in coordination with other components to manage the URL state easily.

@mdwragg could you review? Thanks.

### CODE REVIEW

File: px-app-route/px-app-route-helpers.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
+(function() {
+  var PxAppRouteHelpers = {
+    isDeepEqual: function(a1, a2) {
+      if (!Array.isArray(a1) || !Array.isArray(a2)) return false;
+      if (a1.length !== a2.length) return false;
+
+      for (let i=0; i<a1.length; i++) {
```
Comment (mdwragg) [inline px-app-route/px-app-route-helpers.js]: Do we want to flag this as an es6 src file (given the use of `let`)?

File: px-app-route/px-app-route-helpers.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      var prefixFirstPath = typeof options.prefixFirstPath === 'boolean' ? options.prefixFirstPath : true;
+      var path = '';
+
+      if (Array.isArray(routes.navRoute) && routes.navRoute.length > 0) {
+        path += this.encodePath(routes.navRoute, pathSegment, prefixFirstPath);
+      }
+      if (Array.isArray(routes.assetRoute) && routes.assetRoute.length > 0) {
+        var assetPath = this.encodePath(routes.assetRoute, pathSegment, prefixFirstPath);
+        if (assetPath.length > 0) {
+          path += modelSegment + assetPath;
```
Comment (mdwragg) [inline px-app-route/px-app-route-helpers.js]: If we do es6 this, would template literals clean this up any? Not sure of intent etc. but just sayin ;)

File: px-app-route/px-app-route-helpers.js (hunk 1 above)
Comment (davidrleonard) [inline px-app-route/px-app-route-helpers.js]: 👍  good find, just fell back to var for now so we don't have to worry about transpilation for this.

File: px-app-route/px-app-route-helpers.js (hunk 2 above)
Comment (davidrleonard) [inline px-app-route/px-app-route-helpers.js]: see above, just ripped out the `let` and stuck with good old fashioned ES5

**Pull request merged**